File: P1_E3/CarBuilder.java
package P1_E3;

public class CarBuilder extends VehicleBuilder {
    public CarBuilder() {
        createNewVehicle();
    }

    public void buildEngine(String engine) {
        vehicle.engine = engine;
    }

    public void buildWheels() {
        vehicle.wheels = 4;
    }

    public void buildColor(boolean customized, String color) {
        vehicle.color = customized;
        vehicle.color_vehicle = color;
    }

    public void buildAudioSystem(boolean audioSystem) {
        vehicle.audioSystem = audioSystem;
    }

    public void buildTransmission(boolean transmission) {
        vehicle.transmission = transmission;
    }
}

File: P4/CarBuilder.dart
// ignore_for_file: file_names, non_constant_identifier_names, use_super_parameters

import 'VehicleBuilder.dart';

//Tengo que crear la clase CarBuilder abstracta para no implementar algunos metodos de la superclase
class CarBuilder extends VehicleBuilder{
  
//No estoy seguro de si se llama así con super. Faltaría el baseCost
  CarBuilder({required double baseCost}) : super(baseCost: baseCost){
    createNewVehicle();
    vehicle.tipo = "Coche";
  }

  @override
  void buildEngine(String engine){
    vehicle.engine = engine;
  }

  @override
  void buildWheels(){
    vehicle.wheels = 4;
  }

  @override
  void buildAudioSystem(bool audioSystem){
    vehicle.audioSystem = audioSystem;
  }

  @override
  void buildTransmission(bool transmission){
    vehicle.transmission = transmission;
  }
}